typed-racket-lib/typed-racket/private/type-contract.rkt

```diff
@@ -157,15 +160,21 @@
          (begin (set-box! include-extra-requires? #t)
                 (generate-contract-def e ctc-cache sc-cache))))))
 
-(define (change-provide-fixups forms)
-  (define ctc-cache (make-hash))
-  (define sc-cache (make-hash))
+;; These are probably all in a specific place, which could avoid the big traversal
+(define (change-provide-fixups forms  [ctc-cache (make-hash)] [sc-cache (make-hash)])
   (with-new-name-tables
    (for/list ([form (in-list forms)])
-     (cond [(contract-def/provide-property form)
-            (set-box! include-extra-requires? #t)
-            (generate-contract-def/provide form ctc-cache sc-cache)]
-           [else form]))))
+     (syntax-parse form #:literal-sets (kernel-literals)
+       [_
+        #:when (contract-def/provide-property form)
+        (generate-contract-def/provide form ctc-cache sc-cache)]
+       [(module* name #f forms ...)
+        #`(module* name #f 
```

Comment:
Would preserving syntax locations via `syntax/loc` matter for either of these cases?